.github/workflows/gradle.yml

```diff
@@ -37,17 +37,18 @@ jobs:
             java:
               - '**/*.java'
               - '**/*.gradle'
+              - '**/*.kt'
               - 'airbyte-cdk/java/**/*'
 
   run-check:
     needs:
       - changes
     if: needs.changes.outputs.java == 'true'
-    # The gradle check task which we will run is embarrassingly parallelizable.
-    # We therefore run this on a machine with a maximum number of cores.
-    # We pay per time and per core, so there should be little difference in total cost.
-    # The latency overhead of setting up gradle prior to running the actual task adds up to about a minute.
-    runs-on: connector-test-xxlarge
+    # As of now, 16 cores seems to be enough.
+    # Any revision upwards should be based on a performance analysis of gradle scans.
+    # See https://github.com/airbytehq/airbyte/pull/36055 for an example of this,
+    # which explains why which we went down from 64 cores to 16.
+    runs-on: connector-test-large
```

Comment:
see PR description for why this is OK
tl;dr better caching, disable static analysis on test binaries